nx_parallel/utils/chunk.py

```diff
@@ -6,15 +6,23 @@
 __all__ = ["chunks", "get_n_jobs", "create_iterables"]
 
 
-def chunks(iterable, n_chunks):
-    """Yield exactly `n_chunks` chunks from `iterable`, balancing the chunk sizes."""
+def chunks(iterable, n_chunks, max_chunk_size=None):
+    """
+    Yield chunks by splitting the iterable either into chunks of size at
+    most `max_chunk_size`, or into exactly `n_chunks` balanced chunks if
+    `max_chunk_size` is None.
+    """
     iterable = list(iterable)
-    k, m = divmod(len(iterable), n_chunks)
-    it = iter(iterable)
-    for _ in range(n_chunks):
-        chunk_size = k + (1 if m > 0 else 0)
-        m -= 1
-        yield tuple(itertools.islice(it, chunk_size))
+    if max_chunk_size is not None:
+        for chunk in itertools.batched(iterable, max_chunk_size):
+            yield chunk
+    else:
+        k, m = divmod(len(iterable), n_chunks)
+        it = iter(iterable)
+        for _ in range(n_chunks):
+            chunk_size = k + (1 if m > 0 else 0)
+            m -= 1
+            yield tuple(itertools.islice(it, chunk_size))
```

Comment:
no need for an explicit `else`
```suggestion
    k, m = divmod(len(iterable), n_chunks)
    it = iter(iterable)
    for _ in range(n_chunks):
        chunk_size = k + (1 if m > 0 else 0)
        m -= 1
        yield tuple(itertools.islice(it, chunk_size))
```

Also could you please add a test for the new `max_chunk_size` [here](https://github.com/networkx/nx-parallel/blob/22e4c3cd6541528dff28c991821d2909fc105b1d/nx_parallel/utils/tests/test_chunk.py#L46)?

Reply:
> no need for an explicit `else`

Removing the else would result in double yields, am I missing something here?